src/components/NumberInput/NumberInput.tsx

```diff
@@ -167,7 +168,7 @@ export const NumberInput = ({
 
     const handleOnChange = useCallback(() => {
         const valueWithoutSuffix = getCurrentValueWithoutSuffix();
-        onChange?.(Number(valueWithoutSuffix));
+        onChange?.(valueWithoutSuffix ? Number(valueWithoutSuffix) : undefined);
```

Comment:
why dont you pass it just 0? and could it be possible you dont need to change the types?

Reply:
Maybe I asked wrongly, the question was for undefined. Correct me if I am wrong but if you pass undefined to the onChange function it will turn into 0?

Reply:
That is 100% correct if the component stays as it is. But that is why this PR is required. We have to allow `undefined` to avoid it returning `0` since this is a `number` input.

Reply:
@ryancarville got you, thnx for explanation

Reply:
Anytime. Thank you for your review!